soc/nordic/common/poweroff.c

```diff
@@ -6,6 +6,7 @@
 #include <zephyr/sys/poweroff.h>
 #include <zephyr/toolchain.h>
 
+#if !defined(CONFIG_SOC_SERIES_NRF54HX)
```

Comment:
This `#if` shows that this file is not "common" for Nordic SoCs, so it should be moved to another directory than soc/nordic/common.